Exercise_4.py

```diff
@@ -1,23 +1,44 @@
 # Python program to insert element in binary tree  
+# Time Complexity : O(height)
+# Space Complexity : O(n)
+# Any problem you faced while coding this : No
 class newNode():  
-  
+
     def __init__(self, data):  
         self.key = data 
         self.left = None
         self.right = None
-          
+
 """ Inorder traversal of a binary tree"""
-def inorder(temp): 
-  
-    
-  
-  
+#Inorder traversal follos the simple leeft->root->right strategy hence used the same to traverse the tree recursively.
+def inorder(temp):
+    if temp is None:
+        return
+    inorder(temp.left)
+    print("Data:",temp.key)
+    inorder(temp.right)
+
+
 """function to insert element in binary tree """
+#To insert the new node into the tree, I used BFS to find where a node is missing.
+#When I found a space where either left or right is None, I inserted the node in the tree respectively.
 def insert(temp,key): 
-  
-    
-  
-       
+    bfsList = list()
+    bfsList.append(temp)
+    while(len(bfsList)!=0):
+        elem = bfsList.pop(0)
```

Comment:
Causes shifting of all elements to the left, affects the time complexity, Please use a queue or dequeue instead.

Reply:
Noted